scripts/send-scan.py

```diff
@@ -0,0 +1,128 @@
+                    "installed_version": vuln["InstalledVersion"],
+                    "fixed_version": vuln["FixedVersion"],
+                    "title": vuln["Title"],
+                    "description": vuln["Description"],
+                    "references": "\n".join(vuln["References"]),
+                    "primary_url": vuln["PrimaryURL"],
+                }
+            )
+
+    return record_list
+
+
+def parse_sarif(filename):
+    """Parse SARIF file"""
+    record_list = []
+    with open(filename, "r") as json_file:
+        data = json.load(json_file)
+    if "runs" not in data and "tool" not in data["runs"][0]:
+        # no scan results found, skip this report
+        print(f"No results in report {filename}")
+        return []
+
+    rules = data["runs"][0]["tool"]["driver"]["rules"]
+    results = data["runs"][0]["results"]
+
+    for result in results:
+        record_result = result
+        record_rule = rules[result["ruleIndex"]]
+
+        record_message = record_result["message"]["text"].split("\n")
+        pkg_name = record_message[0].replace("Package: ", "")
+        artifact = os.path.basename(filename).replace(".sarif", "")
+        record_name = str(result["ruleId"] + "-" + artifact + "-" + pkg_name)
+        record_list.append(
+            {
+                "name": record_name,
+                "artifact": artifact,
+                "severity": record_message[3].replace("Severity: ", ""),
+                "cve_id": result["ruleId"],
+                "package_name": pkg_name,
+                "installed_version": record_message[1].replace("Installed Version: ", ""),
+                "fixed_version": record_message[4].replace("Fixed Version: ", ""),
+                "title": record_rule["shortDescription"]["text"],
+                "description": record_rule["help"]["text"],
+                "references": "N/A",
+                "primary_url": record_rule["helpUri"],
+            }
+        )
+
+    return record_list
+
+
+def main(report_path, jira_url):
+    input_path = Path(report_path)
+
+    file_list = []
+    if input_path.is_dir():
+        # directory is supplied, retrieve list of files
+        file_list = list(input_path.iterdir())
+    elif input_path.is_file():
+        file_list.append(input_path)
+
+    for file in file_list:
+        print(f"Processing report in: {file}")
+        if file.suffix == ".json":
+            records = parse_json(file)
+        elif file.suffix == ".sarif":
+            records = parse_sarif(file)
+        else:
+            print(f"Unsupported file type: {file}. Skip it.")
+            continue
+
+        # send records
+        for record in records:
+            requests.post(jira_url, json=record)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--report-path")
+    parser.add_argument("--jira-url")
+    args = parser.parse_args()
+    main(args.report_path, args.jira_url)
```

Comment:
This isn't too complicated, but maybe still some explanation is needed.  Can we add a readme.md that explains how this is to be used?  I'm thinking something like [here](https://github.com/canonical/kubeflow-ci/tree/kf-942-gh188-feat-auto-build-and-scan/scripts/release-charms) (not necessarily as detailed, but a basic usage guideline and why someone should care)